pyinstrument/__main__.py

```diff
@@ -412,9 +412,41 @@ def store_and_consume_remaining(
 def compute_render_options(
     options: CommandLineOptions, renderer_class: type[renderers.Renderer], output_file: TextIO
 ) -> dict[str, Any]:
+    """
+    Given a list of CommandLineOoptionsCompute the rendering options compute the
+    rendering options for the given renderer.
+
+    Raises if there is an error parsing the options.
+
+    """
+
+    unicode_support: bool = file_supports_unicode(output_file)
+    color_support: bool = file_supports_color(output_file)
+
+    error, render_options = _compute_render_options(
+        options, renderer_class, unicode_support, color_support
+    )
+    if error is not None:
+        raise OptionsParseError(error)
+    assert render_options is not None
+    return render_options
+
+
+def _compute_render_options(
+    options: CommandLineOptions,
+    renderer_class: type[renderers.Renderer],
+    unicode_support: bool,
+    color_support: bool,
+) -> Tuple[Optional[str], Optional[dict[str, Any]]]:
+    """
+    Similar as compute_render_options, but return a tuple (error message, data)
+    if there is an error; this will let us reuse _compute_render_options in magics.
+
+    output_file, has been replaced by unicode_support:bool, color_support:bool
+    """
```

Comment:
I think we don't need this extra wrapper function, rather, instead we could wrap the invocations of `compute_render_options` in magic.py with `try:...catch:`...? unless i'm missing something?

Reply:
I just wanted to avoid contaminating the magic system with `OptionsParseError`, which is defined in `__main__`, and I don't want cyclic imports; but I can make `compute_render_option` not raise as it is used in a single place, and raise `OptionsParseError` in the sole place it it used.

Reply:
Also `compute_render_options`, takes an `output_file` while the private variant takes two booleans (unicode_support: bool, color_support: bool), as there are no output file when running the magic. I can lift both of those as well

Reply:
Ah, the unicode/color support bit makes sense. Let's change the `compute_render_options` to use `unicode_support` and `color_support` (moving the file-related stuff to the call site) and then remove the `_compute_render_options` private version. 

I'd rather do the ideomatic thing and raise exceptions on errors. I don't see a problem with `magic.py` importing OptionsParseError, it's already importing from `__main__` anyway. But if it is an issue, we could make it a ValueError and catch that instead.